stablehlo/reference/Ops.cpp

```diff
@@ -850,6 +862,61 @@ Tensor evalFloorOp(const Tensor &operand, ShapedType resultType) {
   return result;
 }
 
+Tensor evalGatherOp(const Tensor &operand, const Tensor &startIndices,
```

Comment:
Very nice to see the implementation so close to the spec. In https://github.com/ghpvnist/stablehlo/pull/5, I made it even closer. PTAL.

Reply:
> Why do we need a clamp here if we can just do if (operandIndex < operand.getShape()) below? The alternative is obviously correct, whereas it wasn't obvious to me what to make of this code.

You're right that we can just do if (operandIndex < operand.getShape()) and it will still be a correct reference implementation as the spec says that the behavior is implementation defined if operandIndex is OOB. However, JAX assumes the clamping behavior, which I chose to implement (and to enable these tests).

Reply:
I would say that we shouldn't enable tests that depend on implementation-defined behavior.

How about in this PR we do `if (operandIndex < operand.getShape())`, and then in a follow-up PR we update spec.md to clamp and enable the rest of the tests? In principle, this all can be done in this PR, but I think it would be better to compartmentalize this change.

Reply:
For example, we did the same for SignOp - first, implemented the spec as is, and then clarified it, updated the interpreter and enabled more tests: https://github.com/openxla/stablehlo/pull/1395.

Reply:
"is also implementation specific right". Agreed - personally, I prefer it because it's probably the simplest way to handle the out of bounds situation.

Reply:
The reason why we need clamping though is that that `Index operandIndex = fullStartIndex + fullOffsetIndex;` could result in correct (jax behavior) bounds when clamped as they can get "too" OOB otherwise. There is a maximum limit to how large `fullOffsetIndex` can get (limited by `resultIndex`), but it is arbitrarily large for fullStartIndex which derives from `startIndices` operand.

Reply:
Huh this was totally non-obvious to me. Looks like this behavior is not covered by the spec at all then.

Reply:
Updated the PR now. One of the interpreter test also fails, so I've removed it for now.